tests/ga/test_extension.py

```diff
@@ -769,6 +769,120 @@ def test_extensions_deleted(self, *args):
         self._assert_handler_status(protocol.report_vm_status, "Ready", 1, "1.0.1")
         self._assert_ext_status(protocol.report_ext_status, "success", 0)
 
+    @patch('subprocess.Popen.poll')
+    def test_install_failure(self, patch_poll, *args):
+        """
+        When extension install fails, the operation should not be retried.
+        """
+        test_data = WireProtocolData(DATA_FILE_EXT_SINGLE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+
+        # Ensure initial install is unsuccessful
+        patch_poll.call_count = 0
+        patch_poll.return_value = 1
+        exthandlers_handler.run()
```

Comment:
I do not think this test is testing what you think it is testing.

`exthandlers_handler.run()` is called inside a `while self.running` loop inside UpdateHandler's `run` method. We should be testing that method to ensure that #840 is fixed.